packages/sandcastle/buildGallery.js

```diff
@@ -0,0 +1,111 @@
+import { readFileSync, existsSync, writeFileSync } from "fs";
+import { parse } from "yaml";
+import { globbySync } from "globby";
+import { basename, dirname, join } from "path";
+import { exit } from "process";
+
+// TODO: probably need to find a way to integrate this into the vite process to make sure it's
+// built during build
+// May also want to move this out of the sandcastle project so we don't have to duplicate
+// files for the local build, need to think about deploying too...
+
+const galleryDirectory = "./public/gallery";
+
+const yamlFiles = globbySync([
+  `${galleryDirectory}/*/sandcastle.yaml`,
+  `!${galleryDirectory}/list.json`,
+]);
```

Comment:
Going with the `globby` approach here instead of just listing the directories lets us "anchor" the existence of a Gallery example around a `sandcastle.yaml` file.